test/test_helper.rb

```diff
@@ -51,5 +51,18 @@ def log_hook(obj)
         ::Pwnlib::Logger.log.instance_variable_set(:@logdev, old)
       end
     end
+
+    # Hooks +$stdin+.
+    #
+    # @param [IO] io
+    #   IO object to be used as +$stdin+.
+    #
+    # @return [Object]
+    #   Returns what the block returned.
+    def hook_stdin(io)
+      org_stdin = $stdin
+      $stdin = io
+      yield.tap { $stdin = org_stdin }
```

Comment:
Why using `tap` here? Should we enclose the it with a `begin...ensure...end` block like the `log_hook()` above?